assembl/tests/model_tests/test_discussion.py

```diff
@@ -1,21 +1,22 @@
 from assembl.models import Discussion, LangString
 
 
-def test_add_discussion(test_session):
-    d = Discussion(
-        topic=u"Education", slug="education",
-        subscribe_to_notifications_on_signup=False,
-        creator=None,
-        session=test_session)
-    d.discussion_locales = ['en', 'fr', 'de']
-    d.terms_and_conditions = LangString.create(
-        u"Use the haptic JSON system, then you can quantify the cross-platform capacitor!", "en")
-    d.legal_notice = LangString.create(
-        u"Use the optical SCSI microchip, then you can generate the cross-platform pixel!", "en")
-    assert d.topic == u"Education"
-    assert d.discussion_locales == ['en', 'fr', 'de']
-    assert d.terms_and_conditions.entries[0].value == u"Use the haptic JSON system, then you can quantify the cross-platform capacitor!"  # noqa: E501
-    assert d.legal_notice.entries[0].value == u"Use the optical SCSI microchip, then you can generate the cross-platform pixel!"  # noqa: E501
+def test_add_discussion(discussion):
+    # d = Discussion(
```

Comment:
I don't like to test a content created by the fixture because the behavior you want to test here is the discussion creation. I think you could have just add the code that deletes the discussion at the end of this test to fix the test isolation problem. I guess it is not a big deal and it is just a personal opinion @vincentfretin ? @ayyazdaniaryan ?
(Also, if in the end you merge it like this, please remove the commented lines here)

Reply:
Indeed, you could have fixed it by removing all created objects I think. You should do a flush before the asserts otherwise you don't really test anything, you will miss the required fields errors for example.